flowinfra: remove isVectorized field from FlowBase

yuzefovich opened the pull request:
**flowinfra: remove isVectorized field from FlowBase**

Previously, FlowBase stored the boolean for whether it is a part of the
vectorized flow. This is not necessary since two kinds of flows can
implement Flow.IsVectorized method themselves.

Release note: None

**colrpc: adjust the test to use coldata.ZeroBatch**

Previously, one of the tests was accumulating all batches assuming that
they were schema-full. But it is no longer true when coldata.ZeroBatch
is emitted, and now this is fixed.

Release note: None

yuzefovich committed:
flowinfra: remove isVectorized field from FlowBase

Previously, FlowBase stored the boolean for whether it is a part of the
vectorized flow. This is not necessary since two kinds of flows can
implement Flow.IsVectorized method themselves.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42868)
<!-- Reviewable:end -->

rafiss reviewed (approved):
> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42868)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+

yuzefovich committed:
colrpc: adjust the test to use coldata.ZeroBatch

Previously, one of the tests was accumulating all batches assuming that
they were schema-full. But it is no longer true when coldata.ZeroBatch
is emitted, and now this is fixed.

Release note: None

yuzefovich commented:
The build failed because of the other PR. I added a second commit here to fix the test.

yuzefovich commented:
bors r+
